var/spack/repos/builtin/packages/jsonnet/package.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack import *
+
+
+class Jsonnet(MakefilePackage):
+    """A data templating language for app and tool developers based on JSON"""
+
+    homepage = "https://jsonnet.org/"
+    git = "https://github.com/google/jsonnet.git"
```

Comment:
Can we download from a URL instead of from git? That allows us to checksum the tarball.

Reply:
Attempted in latest push.

Reply:
You can add a global `url` attribute so that you don't need to specify the URL for each version.

Reply:
The folks I see using `url` have explicit versions in there... are they transformed automatically or?

Reply:
They are, Spack knows how to extract and substitute versions from URLs:
```console
$ spack url parse https://github.com/google/jsonnet/archive/refs/tags/v0.18.0.tar.gz
==> Parsing URL: https://github.com/google/jsonnet/archive/refs/tags/v0.18.0.tar.gz

==> Matched version regex  1: r'^v?(\\d[\\d._-]*)$'
==> Matched  name   regex  0: r'github\\.com/[^/]+/([^/]+)'

==> Detected:
    https://github.com/google/jsonnet/archive/refs/tags/v0.18.0.tar.gz
                              -------                    ~~~~~~
    name:    jsonnet
    version: 0.18.0

==> Substituting version 9.9.9b:
    https://github.com/google/jsonnet/archive/refs/tags/v9.9.9b.tar.gz
                              -------                    ~~~~~~
```

Reply:
Excellent, I'll push up a change.